main.py

```diff
@@ -212,10 +243,27 @@ def main():
         else:
             model_kwargs["torch_dtype"] = dict_precisions[args.precision]
 
-        model = AutoModelForCausalLM.from_pretrained(
-            args.model,
-            **model_kwargs,
-        )
+            if args.max_memory_per_gpu:
+                model_kwargs["max_memory"] = get_gpus_max_memory(args.max_memory_per_gpu)
+                model_kwargs["offload_folder"] = "offload"
+                model_kwargs["device_map"] = "auto"
+
+        
+        if args.modeltype == "causal":
+            model = AutoModelForCausalLM.from_pretrained(
+                args.model,
+                **model_kwargs,
+            )
+        elif args.modeltype == "seq2seq":
+            print("WARNING: seq2seq models have only been tested for HumanEvalPack & CodeT5+ models.")
```

Comment:
for consistency with other warnings (we still need to import `warnings`)
```suggestion
            warnings.warn("Seq2Seq models have only been tested for HumanEvalPack & CodeT5+ models.")
```